### PR TITLE
Keywords can now repeat multiple times in same sentence

### LINKED ISSUE #88: Keywords can't repeat
If the same keyword is in a sentence twice, only the first one gets repeated. Users probably won't notice this immediately, low priority.

### PR DESCRIPTION
**Pull request by benfrancis314**
Also more efficient keyword finding process. Insight is that when we know a word in a sentence is a keyword, and we look to find it with indexOf(), we shouldn't start at the beginning, but rather at the end of the last search. Since it looks through the words in a sentence sequentially, the next keyword is guaranteed to be after the previous one. If there are two keywords in a sentence, this also allows it to find both; it is searching after the first keyword, so it can't find it again. 

Close #88

### CODE REVIEW

File: extension/content_script.js [hunk 1]
```diff
@@ -298,12 +298,15 @@ function highlightKeyWords(container, start, end) {
 	let wordRegex = /\b\w+\b/g;
 	let wordList = sentenceText.match(wordRegex);
 	let keywords = doc.getKeyWords();
+	// Look where keyword is in sentence AFTER last search. Guaranteed to be after. Init to 0. 
+	let keyword_search_start_pointer = 0; 
 	for (var i in wordList) { // Accentuate keywords
 		// TODO: This only gets the first occurence of each word in the sentence; should get all
```
Comment (sjonany) [inline extension/content_script.js]: Remove the TODO?

File: extension/content_script.js [hunk 2]
```diff
@@ -298,12 +298,15 @@ function highlightKeyWords(container, start, end) {
...
 		let word = wordList[i];
 		if (keywords.has(word.toLowerCase())) { // See if each word is a keyword
-			var word_start = containerText.indexOf(word, start);
+			var word_start = containerText.indexOf(word, start+keyword_search_start_pointer);
 			var word_len = word.length;
+			keyword_search_start_pointer += word_len;
```
Comment (sjonany) [inline extension/content_script.js]: I think this should be word_start + word_len.

Concerns w/ current approach:
1. Undercount the actual search_start_pointer, cos it doesn't account for the word lens of the non-matching words.
E.g. 
"match abasdfasdfasdf match"
I think by the second "match" ur keyword_ssearch_start_poniter is pointing to somewhere in the middle of the random string

2. undercounting cos ur not counting the word boundary chars.

**Review by sjonany — CHANGES_REQUESTED** [review]

File: extension/content_script.js (hunk 2 above)
Comment (benfrancis314) [reply extension/content_script.js]: Good reasoning

File: extension/content_script.js [hunk 3]
```diff
@@ -298,12 +298,14 @@ function highlightKeyWords(container, start, end) {
...
+	// Look where keyword is in sentence AFTER last search. Guaranteed to be after. Init to 0. 
+	let keyword_search_start_pointer = start; 
 	for (var i in wordList) { // Accentuate keywords
-		// TODO: This only gets the first occurence of each word in the sentence; should get all
 		let word = wordList[i];
 		if (keywords.has(word.toLowerCase())) { // See if each word is a keyword
-			var word_start = containerText.indexOf(word, start);
+			var word_start = containerText.indexOf(word, keyword_search_start_pointer);
 			var word_len = word.length;
+			keyword_search_start_pointer = word_start + word_len;
```
Comment (sjonany) [inline extension/content_script.js]: nit: save a line and just do + word.length

File: extension/content_script.js [hunk 4]
```diff
@@ -298,12 +298,14 @@ function highlightKeyWords(container, start, end) {
 	let wordRegex = /\b\w+\b/g;
 	let wordList = sentenceText.match(wordRegex);
 	let keywords = doc.getKeyWords();
+	// Look where keyword is in sentence AFTER last search. Guaranteed to be after. Init to 0.
```
Comment (sjonany) [inline extension/content_script.js]: nit: Remove init to 0?

**Review by sjonany — CHANGES_REQUESTED** [review]
LGTM, feel free to squash and merge after addressing nits

File: extension/content_script.js (hunk 3 above)
Comment (benfrancis314) [reply extension/content_script.js]: We use word_len below as well: 

`container.markRanges([{ 
			start: word_start,
			length: word_len
		}], {`

Comment (benfrancis314) [reply extension/content_script.js]: Ehh I can still get rid of it though, it's self explanatory

**Review by sjonany — APPROVED** [review]

**Pull request merged**